compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -816,6 +823,7 @@ impl Context {
             }
             Type::Slice(_) => {
                 // TODO(#2752): need SSA values here to fetch the len like we do for a Type::Array
+                // Update this to enable fetching slices from nested arrays
```

Comment:
Going to make a new issue for this as dynamically indexing a non-homogenous slice is different than returning a slice from a non-homogenous array/slice. I will see how bad it will be to add this, but I think it should still be scoped separately from this PR.